source/nodejs/adaptivecards/src/card-elements.ts

```diff
@@ -975,6 +980,34 @@ export class TextBlock extends CardElement {
     }
 }
 
+class Label extends TextBlock {
+    private static uniqueIdCounter = 0;
+
+    protected getRenderedDomElementType(): string {
+        return "label";
+    }
+
+    protected internalRender(): HTMLElement {
+        let renderedElement = <HTMLLabelElement>super.internalRender();
+
+        if (!Utils.isNullOrEmpty(this.forElementId)) {
+            renderedElement.htmlFor = this.forElementId;
+        }
+
+        return renderedElement;
+    }
+
+    forElementId: string;
+
+    static generateUniqueInputId(): string {
+        let uniqueId = "__ac-input-id" + Label.uniqueIdCounter;
```

Comment:
I'm not sure how valid this concern is but since IDs need to be unique throughout the entire DOM I wonder if we run the risk of conflicts for any cards that may be delivered via server-side HTML and then appended using the JS lib? in the .NET HTML renderer I did a Guid.NewGuid substring 8 to try reduce the chance of conflict 
https://github.com/Microsoft/AdaptiveCards/blob/5577cd2802fb0fa21f2319c6cb6229bed55a0a0d/source/dotnet/Library/AdaptiveCards.Rendering.Html/AdaptiveCardRenderer.cs#L23

Seems JS doesn't have such a facility but something like this function may work well enough:
https://stackoverflow.com/questions/105034/create-guid-uuid-in-javascript